lib/puppet/type/ecs_service.rb

```diff
@@ -0,0 +1,56 @@
+Puppet::Type.newtype(:ecs_service) do
+  @doc = 'Type representing ECS services.'
+
+  ensurable
+
+  autorequire(:ecs_cluster) do
+    self[:cluster]
+  end
+
+  autorequire(:ecs_task_definition) do
+    self[:task_definition]
+  end
+
+  newparam(:name, namevar: true) do
+    desc 'The name of the cluster to manage.'
```

Comment:
You can generate markdownish formatted docs with `RUBYLIB=lib puppet doc -r type` in the module and copy/paste that to the readme to get started. Each property & param needs a `desc` for this to work though.